src/main/java/org/onedatashare/server/service/JobService.java

```diff
@@ -32,7 +32,9 @@
     }
 
     public Mono<List<Job>> getAllJobsForUser(String cookie) {
-        return userService.getJobs(cookie).flatMap(this::getJobByUUID).publishOn(Schedulers.parallel()).collectList();
+        return userService.getJobs(cookie).flatMap(this::getJobByUUID).publishOn(Schedulers.parallel())
+                .filter(job -> !job.deleted)
```

Comment:
I suppose we had discussed the cons of this method. Please change the implementation to push the filtering to MongoDB.

Reply:
Implemented suggested changes. Please review.